test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -80,6 +80,34 @@ def test_deploy_succeeded_with_max_unavailable
     refute deploy.deploy_succeeded?
   end
 
+  def test_deploy_succeeded_with_annotation_as_number
+    deployment_status = {
+      "replicas" => 3, # one terminating in old rs, one starting in new rs, one up in new rs
+      "updatedReplicas" => 2,
+      "unavailableReplicas" => 2,
+      "availableReplicas" => 1
+    }
+
+    rs_status = {
+      "replicas" => 2,
+      "availableReplicas" => 1,
+      "readyReplicas" => 1
+    }
+    replica_sets = [build_rs_template(status: rs_status)]
+
+    deploy = build_synced_deployment(
+      template: build_deployment_template(status: deployment_status, rollout: '33%', max_unavailable: 2),
+      replica_sets: replica_sets
+    )
+    assert deploy.deploy_succeeded?
+
+    deploy = build_synced_deployment(
+      template: build_deployment_template(status: deployment_status, rollout: '34%', max_unavailable: 2),
```

Comment:
let's add `0%` and `100%` cases for good measure. We could also leave out `max_unavailable` in this test, since it shouldn't matter. Or we could set it to `0%` to prove it isn't being referenced.